homeassistant/components/nest/__init__.py

```diff
@@ -76,76 +80,44 @@
                 return
 
 
-async def async_request_configuration(nest, hass, config):
-    """Request configuration steps from the user."""
-    configurator = hass.components.configurator
-    if 'nest' in _CONFIGURING:
-        _LOGGER.debug("configurator failed")
-        configurator.async_notify_errors(
-            _CONFIGURING['nest'], "Failed to configure, please try again.")
+async def async_setup(hass, config):
+    """Set up Nest components."""
+    if DOMAIN not in config:
         return
 
-    async def async_nest_config_callback(data):
-        """Run when the configuration callback is called."""
-        _LOGGER.debug("configurator callback")
-        pin = data.get('pin')
-        if await async_setup_nest(hass, nest, config, pin=pin):
-            # start nest update event listener as we missed startup hook
-            hass.async_add_job(async_nest_update_event_broker, hass, nest)
-
-    _CONFIGURING['nest'] = configurator.async_request_config(
-        "Nest", async_nest_config_callback,
-        description=('To configure Nest, click Request Authorization below, '
-                     'log into your Nest account, '
-                     'and then enter the resulting PIN'),
-        link_name='Request Authorization',
-        link_url=nest.authorize_url,
-        submit_caption="Confirm",
-        fields=[{'id': 'pin', 'name': 'Enter the PIN', 'type': ''}]
-    )
-
-
-async def async_setup_nest(hass, nest, config, pin=None):
-    """Set up the Nest devices."""
-    from nest.nest import AuthorizationError, APIError
-    if pin is not None:
-        _LOGGER.debug("pin acquired, requesting access token")
-        error_message = None
-        try:
-            nest.request_token(pin)
-        except AuthorizationError as auth_error:
-            error_message = "Nest authorization failed: {}".format(auth_error)
-        except APIError as api_error:
-            error_message = "Failed to call Nest API: {}".format(api_error)
-
-        if error_message is not None:
-            _LOGGER.warning(error_message)
-            hass.components.configurator.async_notify_errors(
-                _CONFIGURING['nest'], error_message)
-            return False
-
-    if nest.access_token is None:
-        _LOGGER.debug("no access_token, requesting configuration")
-        await async_request_configuration(nest, hass, config)
-        return False
+    conf = config[DOMAIN]
+
+    local_auth.initialize(hass, conf[CONF_CLIENT_ID], conf[CONF_CLIENT_SECRET])
+
+    filename = config.get(CONF_FILENAME, NEST_CONFIG_FILE)
+    access_token_cache_file = hass.config.path(filename)
+
+    if await hass.async_add_job(os.path.isfile, access_token_cache_file):
```

Comment:
Is await unnecessary? I think async_add_job is not a coroutine

Reply:
It returns a task that is awaitable for the result of the function. This is used to run non asyncio functions inside a worker thread pool and get the response.